### PR TITLE
Unsubscribe by type

### LINKED ISSUE #317: SubscriptionSet: Remove all subscriptions by type of objects
remove all subscriptions by the type of realm objects they were created for

### LINKED ISSUE #320: SubscriptionSet: Remove subscription by results instance

### LINKED ISSUE #332: Story: Support SubscriptionSet API

### PR DESCRIPTION
**Pull request by nielsenko**
- Add removeByType
- Add test of removeByType
- Update CHANGELOG

Resolve #317 
Resolve #332

### CODE REVIEW

File: test/subscription_test.dart [hunk 1]
```diff
@@ -379,6 +379,26 @@ Future<void> main([List<String>? args]) async {
...
+    });
+
+    expect(subscriptions.length, 4);
+
+    subscriptions.update((mutableSubscriptions) {
+      mutableSubscriptions.removeByType<Task>();
+    });
+
+    expect(subscriptions, [s]);
+  });
```
Comment (desistefanova) [inline test/subscription_test.dart:400]:
```suggestion
  testSubscriptions('MutableSubscriptionSet.removeByType', (realm) {
    final subscriptions = realm.subscriptions;

    late Subscription scheduleSubscription;
    subscriptions.update((mutableSubscriptions) {
      mutableSubscriptions.add(realm.query<Task>(r'_id == $0', [ObjectId()]));
      mutableSubscriptions.add(realm.query<Task>(r'_id == $0', [ObjectId()]));
      mutableSubscriptions.add(realm.query<Task>(r'_id == $0', [ObjectId()]));
      scheduleSubscription = mutableSubscriptions.add(realm.all<Schedule>());
    });

    expect(subscriptions.length, 4);

    subscriptions.update((mutableSubscriptions) {
      mutableSubscriptions.removeByType<Task>();
    });

    expect(subscriptions, [scheduleSubscription]);
  });
```

Comment (desistefanova) [reply test/subscription_test.dart:400]: I've just renamed the variable.

**Review by desistefanova — APPROVED** [review]
It looks OK.

File: lib/src/subscription.dart [hunk 2]
```diff
@@ -182,7 +182,7 @@ abstract class SubscriptionSet with IterableMixin<Subscription> implements Final
   Subscription operator [](int index) => elementAt(index);
 
   @override
-  _SubscriptionIterator get iterator => _SubscriptionIterator._(this);
+  Iterator<Subscription> get iterator => _SubscriptionIterator._(this);
```
Comment (blagoev) [inline lib/src/subscription.dart:185]: wow, how did we not notice this before

**Review by blagoev — APPROVED** [review]

**Review by nirinchev — APPROVED** [review]

File: lib/src/subscription.dart (hunk 2 above)
Comment (nielsenko) [reply lib/src/subscription.dart:185]: It worked just fine before. It is just not good style to expose private classes on public methods.

**Pull request merged**